poc/runtime/src/xcq.rs

```diff
@@ -17,11 +19,70 @@ struct HostFunctions;
 
 impl poc_executor::XcqExecutorContext for HostFunctions {
     fn register_host_functions<T>(&mut self, linker: &mut poc_executor::Linker<T>) {
-        linker.func_wrap("host_call", || -> u32 { 100 }).unwrap();
+        linker
+            .func_wrap(
+                "query_balances",
+                move |caller: poc_executor::Caller<_>, variant: u32, accounts_ptr: u32, accounts_len: u32| -> u64 {
+                    // variant 0 means free balance
+                    // variant 1 means reserved balance
+                    // variant 2 means free+reserved
+                    let mut sum = 0u64;
```

Comment:
the sum logic should happen in guest program. the host only does the query

Reply:
I'm thinking make host call:
1. `query_balances(variant, account_ids_ptr, bytes_len)`(which make host decode `Vec<AccountId>`'s length)  and return slices of balance (write to guest as return values) 
2. or let guest has the type information, decode length, then do `query_balance(variant, account_id)`
I think the latter is more general and make the host call api simple.

Reply:
host function should be minimal 
it should be query one account id at a time so and we know the size of AccountId and can just hardcode it here